src/main/java/net/snowflake/ingest/streaming/internal/FlushService.java

```diff
@@ -359,11 +317,17 @@ void distributeFlushTasks() {
         itr = this.channelCache.iterator();
     List<Pair<BlobData<T>, CompletableFuture<BlobMetadata>>> blobs = new ArrayList<>();
     List<ChannelData<T>> leftoverChannelsDataPerTable = new ArrayList<>();
+    boolean isBlobCreated = true;
+    String currentBlobPath = null;
 
     while (itr.hasNext() || !leftoverChannelsDataPerTable.isEmpty()) {
       List<List<ChannelData<T>>> blobData = new ArrayList<>();
       float totalBufferSizeInBytes = 0F;
-      final String blobPath = getBlobPath(this.targetStage.getClientPrefix());
+      if (isBlobCreated) {
```

Comment:
you could just call generateBlobPath from outside the while loop in line 321 directly.

can add `if (this.channelCache.size() == 0) { return; }` block at the start of this method to make sure generate is only called when necessary.